flocker/node/agents/ebs.py

```diff
@@ -42,6 +47,93 @@
 MAX_ATTACH_RETRIES = 3
 
 
+class VolumeOperations(Names):
+    """
+    Supported EBS backend operations on a volume.
+    """
+    CREATE = NamedConstant()
+    ATTACH = NamedConstant()
+    DETACH = NamedConstant()
+    DESTROY = NamedConstant()
+
+
+class VolumeStates(Values):
+    """
+    Expected EBS volume states during ``VolumeOperations``.
+    """
+    EMPTY = ValueConstant('')
+    CREATING = ValueConstant(u'creating')
+    AVAILABLE = ValueConstant(u'available')
+    ATTACHING = ValueConstant(u'attaching')
+    IN_USE = ValueConstant(u'in-use')
+    DETACHING = ValueConstant(u'detaching')
+    DELETING = ValueConstant(u'deleting')
```

Comment:
Thats an interesting state. Have never observed ``deleted``, but it could be a transient state right after ``deleting``. Might be worth it to add multiple transient state support per operation. Will take it up as a follow up. Thanks for pointing out!